common/aws/dynamodb/client.go

```diff
@@ -307,8 +318,42 @@ func (c *Client) writeItems(ctx context.Context, tableName string, requestItems
 			}
 		}
 
-		startIndex += dynamoBatchLimit
+		startIndex += dynamoBatchWriteLimit
 	}
 
 	return failedItems, nil
 }
+
+func (c *Client) readItems(ctx context.Context, tableName string, keys []Key) ([]Item, error) {
+	startIndex := 0
+	items := make([]Item, 0)
+	for startIndex < len(keys) {
+		remainingNumKeys := float64(len(keys) - startIndex)
+		batchSize := int(math.Min(float64(dynamoBatchReadLimit), remainingNumKeys))
+		keysBatch := keys[startIndex : startIndex+batchSize]
+		output, err := c.dynamoClient.BatchGetItem(ctx, &dynamodb.BatchGetItemInput{
```

Comment:
We wouldn't fetch that much data most cases. 
In case we do, it will be handled in L350-353 as the output will contain items (`output.Responses`) within the 16MB limit and unprocessed keys will be returned in `output.UnprocessedKeys`